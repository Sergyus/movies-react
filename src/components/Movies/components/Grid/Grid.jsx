import React, { useState, useCallback } from 'react';
import GridCardContainer from './components/GridCard';

const Grid = ({ movies, toggleModal }) => {
  const [activeCard, setActiveCard] = useState(0);
  const changeActiveCard = useCallback(
    (id) => {
      setActiveCard(id);
    },
    [],
  );
  return movies.map(movie => (
    <GridCardContainer
      activeCard={activeCard}
      changeActiveCard={changeActiveCard}
      key={movie.id}
      movie={movie}
      toggleModal={toggleModal}
      isActive={movie.id !== activeCard}
    />
  ));
};

export default Grid;
