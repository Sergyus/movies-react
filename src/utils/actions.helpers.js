export const createAsyncFnRequest = requestName => ({
  type: requestName,
});

export const createAsyncFnFailure = (requestName, error) => ({
  type: requestName,
  error: error.message,
});
