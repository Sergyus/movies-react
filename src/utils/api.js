const MOVIE_DB_API_KEY = "ed98d25919802b5020e8feb6c3923886";
const MOVIE_DB_BASE_URL = "https://api.themoviedb.org/3";
const LANGUAGE = "ru-RU";

const createMovieDbUrl = (relativeUrl, queryParams) => {
  let baseUrl = `${MOVIE_DB_BASE_URL}${relativeUrl}?api_key=${MOVIE_DB_API_KEY}&language=${LANGUAGE}`;
  if (queryParams) {
    Object.keys(queryParams).forEach(paramName => {
      baseUrl += `&${paramName}=${queryParams[paramName]}`;
      return baseUrl;
    });
  }
  return baseUrl;
};

const getLatestMovies = ({ page }) => {
  const fullUrl = createMovieDbUrl("/movie/popular", {
    page
  });
  return fetch(fullUrl);
};

const getTopMovies = ({ page }) => {
  const fullUrl = createMovieDbUrl("/movie/top_rated", {
    page
  });
  return fetch(fullUrl);
};

const getUpcomingMovies = ({ page }) => {
  const fullUrl = createMovieDbUrl("/movie/upcoming", {
    page
  });
  return fetch(fullUrl);
};

const searchMovies = ({ page, query }) => {
  const fullUrl = `${createMovieDbUrl("/search/movie", {
    page,
    query
  })}&include_adult=false`;
  return fetch(fullUrl);
};

const getGenres = () => {
  const fullUrl = createMovieDbUrl("/genre/movie/list");
  return fetch(fullUrl);
};

const getMoviesByGenre = ({ page, genre }) => {
  const fullUrl = `${createMovieDbUrl("/discover/movie", {
    page
  })}&sort_by=popularity.desc&include_adult=false&include_video=false&with_genres=${genre}`;
  return fetch(fullUrl);
};

const getMovieDetails = ({ movieId }) => {
  const fullUrl = createMovieDbUrl(`/movie/${movieId}`);
  return fetch(fullUrl);
};

const getMovieVideo = movieId => {
  const fullUrl = createMovieDbUrl(`/movie/${movieId}/videos`);
  return fetch(fullUrl);
};

export {
  getTopMovies,
  searchMovies,
  getMovieDetails,
  getGenres,
  getMoviesByGenre,
  getLatestMovies,
  getUpcomingMovies,
  getMovieVideo
};
