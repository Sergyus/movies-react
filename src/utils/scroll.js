const getScrollDownPercentage = ({ scrollHeight, scrollPos, clientHeight }) => {
  const currentPosition = scrollPos + clientHeight;
  const percentageScrolled = currentPosition / scrollHeight;
  return percentageScrolled;
};

export default getScrollDownPercentage;
