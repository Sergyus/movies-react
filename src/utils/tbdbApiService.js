import noAvailableImage from '../assets/images/noAvailableImage.png';

const urlBase = 'https://image.tmdb.org/t/p/original';
const urlMinImg = 'https://image.tmdb.org/t/p/w300';

function getNormalRuntime(time) {
  const hours = Math.floor(time / 60);
  const minutes = time - 60 * hours;
  return `${hours}h ${minutes}m`;
}

export function getMovieGenres(genresIds = [], genres = []) {
  const genresArr = [];
  genresIds.map(genre => genres.map((movie) => {
    if (movie.id === genre) {
      return genresArr.push({ id: movie.id, name: movie.name });
    }
    return 0;
  }));
  return genresArr;
}

export function getMovieInfo(movie) {
  return {
    title: movie.title,
    genres: movie.genres,
    rating: movie.vote_average,
    runtime: getNormalRuntime(movie.runtime),
    overview: movie.overview,
    image: movie.backdrop_path ? `${urlBase}${movie.backdrop_path}` : noAvailableImage,
  };
}

export function getMovies(movies = [], allGenres = []) {
  return movies.map(movie => ({
    id: movie.id,
    rating: movie.vote_average,
    title: movie.title,
    genres: getMovieGenres(movie.genre_ids, allGenres),
    overview: movie.overview,
    image: movie.poster_path ? `${urlMinImg}${movie.poster_path}` : noAvailableImage,
  }));
}
