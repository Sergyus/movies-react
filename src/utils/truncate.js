const truncate = (description = '', maxLength) => (
  description.length && description.length > maxLength ? `${description.slice(0, maxLength)}...` : description
);

export default truncate;
