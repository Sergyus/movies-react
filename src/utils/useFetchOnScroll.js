import { useLayoutEffect } from 'react';
import getScrollDownPercentage from './scroll';

export const handleScroll = (
  isLoading, hasMorePages, filter, currentPage,
  getLatestMovies, getTopRatedMovies, getUpcomingMovies,
  getMoviesByGenre, currentGenre, getSearchMovies, currentSearchQuery, window,
) => {
  if (!isLoading) {
    const { scrollHeight, clientHeight } = window.document.documentElement;
    const scrollPos = window.pageYOffset;
    const percentageScrolled = getScrollDownPercentage(
      { scrollHeight, clientHeight, scrollPos },
    );
    if (percentageScrolled > 0.9 && hasMorePages) {
      switch (filter) {
        case 'Trending':
          getLatestMovies(currentPage + 1);
          break;
        case 'Top Rated':
          getTopRatedMovies(currentPage + 1);
          break;
        case 'Coming Soon':
          getUpcomingMovies(currentPage + 1);
          break;
        case 'Genres':
          getMoviesByGenre(currentPage + 1, currentGenre);
          break;
        case 'Search':
          getSearchMovies(currentPage + 1, currentSearchQuery);
          break;
        default: break;
      }
    }
  }
};
export const useFetchOnScroll = (...args) => {
  useLayoutEffect(() => {
    const callback = () => handleScroll(...args, window);
    document.addEventListener('scroll', callback);
    return () => {
      document.removeEventListener('scroll', callback);
    };
  }, [...args]);
};
