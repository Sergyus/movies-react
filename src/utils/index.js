import {
  getTopMovies, searchMovies, getGenres, getMoviesByGenre,
  getLatestMovies, getUpcomingMovies, getMovieVideo,
} from './api';
import { getMovies } from './tbdbApiService';
import truncate from './truncate';
import { useFetchOnScroll } from './useFetchOnScroll';
import useBodyScrollLock from './useBodyScrollLock';
import useOnClickOutside from './useOnClickOutside';
import { createAsyncFnFailure, createAsyncFnRequest } from './actions.helpers';

export {
  getTopMovies, searchMovies, getGenres, getMoviesByGenre,
  getLatestMovies, getUpcomingMovies, getMovieVideo,
  truncate, useBodyScrollLock, useOnClickOutside, getMovies,
  createAsyncFnFailure, createAsyncFnRequest, useFetchOnScroll,
};
