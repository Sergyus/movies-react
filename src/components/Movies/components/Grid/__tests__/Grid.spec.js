import React from 'react';
import { create, act } from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Grid from '..';
import { getMovies } from '../../../../../utils';
import mockMovies from '../../../../../assets/json/popular.json';
import { genres } from '../../../../../assets/json/genres.json';

describe('Grid component renders correctly', () => {
  const movies = getMovies(mockMovies.results, genres);
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {
    app: {
      movies: [],
      genres: [],
      isLoading: true,
      currentPage: 1,
      error: false,
      filter: 'Trending',
      currentGenre: '',
      currentSearchQuery: '',
      videoKey: '',
      currentTotalPages: 1,
    },
    ui: {
      layout: 'grid',
    },
  };
  const store = mockStore(initialState);
  it('Grid renders correctly', () => {
    const tree = create(
      <Provider store={store}>
        <Grid movies={movies} />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Change active card called correctly', () => {
    const tree = create(
      <Provider store={store}>
        <Grid movies={movies} />
      </Provider>,
    );
    const mockFunction = jest.fn(tree.toTree().rendered.props.changeActiveCard);
    act(() => {
      mockFunction();
    });
    expect(mockFunction).toHaveBeenCalled();
  });
  it('Change active card works correctly', () => {
    const tree = create(
      <Provider store={store}>
        <Grid movies={movies} />
      </Provider>,
    );
    const mockFunction = jest.fn(
      id => tree.toTree().rendered.rendered[0].props.changeActiveCard(id),
    );
    act(() => {
      mockFunction(123);
    });
    expect(mockFunction).toHaveBeenCalled();
  });
});
