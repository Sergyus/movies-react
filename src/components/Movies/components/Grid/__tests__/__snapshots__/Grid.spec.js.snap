// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component renders correctly Grid renders correctly 1`] = `
Array [
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/AnLCqcH66Ecu8VFmXu3RofK4SMK.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          T-34
        </h3>
        <p>
          <span
            className="genre"
          >
            War
          </span>
          <span
            className="genre"
          >
            Drama
          </span>
          <span
            className="genre"
          >
            Adventure
          </span>
          <span
            className="genre"
          >
            Action
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        4.5
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/svIDTNUoajS8dLEo7EosxvyAsgJ.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Glass
        </h3>
        <p>
          <span
            className="genre"
          >
            Thriller
          </span>
          <span
            className="genre"
          >
            Mystery
          </span>
          <span
            className="genre"
          >
            Drama
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        6.8
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/AiRfixFcfTkNbn2A73qVJPlpkUo.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Robin Hood
        </h3>
        <p>
          <span
            className="genre"
          >
            Adventure
          </span>
          <span
            className="genre"
          >
            Action
          </span>
          <span
            className="genre"
          >
            Thriller
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        5.8
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Bohemian Rhapsody
        </h3>
        <p>
          <span
            className="genre"
          >
            Drama
          </span>
          <span
            className="genre"
          >
            Music
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        8.2
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/l76Rgp32z2UxjULApxGXAPpYdAP.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Overlord
        </h3>
        <p>
          <span
            className="genre"
          >
            Horror
          </span>
          <span
            className="genre"
          >
            War
          </span>
          <span
            className="genre"
          >
            Action
          </span>
          <span
            className="genre"
          >
            Science Fiction
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        6.9
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/8yZAx7tlKRZIg7pJfaPhl00yHIQ.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Escape Room
        </h3>
        <p>
          <span
            className="genre"
          >
            Horror
          </span>
          <span
            className="genre"
          >
            Thriller
          </span>
          <span
            className="genre"
          >
            Action
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        6.3
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Aquaman
        </h3>
        <p>
          <span
            className="genre"
          >
            Action
          </span>
          <span
            className="genre"
          >
            Fantasy
          </span>
          <span
            className="genre"
          >
            Science Fiction
          </span>
          <span
            className="genre"
          >
            Adventure
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        6.8
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/uXJVpPXxZO4L8Rz3IG1Y8XvZJcg.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Mortal Engines
        </h3>
        <p>
          <span
            className="genre"
          >
            Science Fiction
          </span>
          <span
            className="genre"
          >
            Action
          </span>
          <span
            className="genre"
          >
            Adventure
          </span>
          <span
            className="genre"
          >
            Fantasy
          </span>
          <span
            className="genre"
          >
            Thriller
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        6
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/v3QyboWRoA4O9RbcsqH8tJMe8EB.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Creed II
        </h3>
        <p>
          <span
            className="genre"
          >
            Drama
          </span>
          <span
            className="genre"
          >
            Action
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        6.7
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/ij0xoc13hGhrYIlXGGuPXWTh3vi.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          How to Train Your Dragon: The Hidden World
        </h3>
        <p>
          <span
            className="genre"
          >
            Animation
          </span>
          <span
            className="genre"
          >
            Family
          </span>
          <span
            className="genre"
          >
            Adventure
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        8.1
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          A Star Is Born
        </h3>
        <p>
          <span
            className="genre"
          >
            Drama
          </span>
          <span
            className="genre"
          >
            Music
          </span>
          <span
            className="genre"
          >
            Romance
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        7.4
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/t0idiLMalKMj2pLsvqHrOM4LPdQ.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          The Mule
        </h3>
        <p>
          <span
            className="genre"
          >
            Crime
          </span>
          <span
            className="genre"
          >
            Drama
          </span>
          <span
            className="genre"
          >
            Thriller
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        6.3
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Avengers: Infinity War
        </h3>
        <p>
          <span
            className="genre"
          >
            Adventure
          </span>
          <span
            className="genre"
          >
            Action
          </span>
          <span
            className="genre"
          >
            Fantasy
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        8.3
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/fw02ONlDhrYjTSZV8XO6hhU3ds3.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Bumblebee
        </h3>
        <p>
          <span
            className="genre"
          >
            Action
          </span>
          <span
            className="genre"
          >
            Adventure
          </span>
          <span
            className="genre"
          >
            Science Fiction
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        6.5
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/f27RTnWEc2bhVtfSlKmxxypznfR.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Widows
        </h3>
        <p>
          <span
            className="genre"
          >
            Crime
          </span>
          <span
            className="genre"
          >
            Drama
          </span>
          <span
            className="genre"
          >
            Thriller
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        6.7
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/hgWAcic93phg4DOuQ8NrsgQWiqu.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Serenity
        </h3>
        <p>
          <span
            className="genre"
          >
            Thriller
          </span>
          <span
            className="genre"
          >
            Drama
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        5.1
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Venom
        </h3>
        <p>
          <span
            className="genre"
          >
            Science Fiction
          </span>
          <span
            className="genre"
          >
            Action
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        6.6
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/hPuZR2bFoV0kuBrA98exWyvTf40.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Can You Ever Forgive Me?
        </h3>
        <p>
          <span
            className="genre"
          >
            Drama
          </span>
          <span
            className="genre"
          >
            Comedy
          </span>
          <span
            className="genre"
          >
            Crime
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        7.1
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/t0jwlwFUrtkPJoq5JH9YrRVROXZ.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          Fate/stay night: Heaven’s Feel II. lost butterfly
        </h3>
        <p>
          <span
            className="genre"
          >
            Animation
          </span>
          <span
            className="genre"
          >
            Action
          </span>
          <span
            className="genre"
          >
            Fantasy
          </span>
          <span
            className="genre"
          >
            Adventure
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        3.9
      </span>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      height="70%"
      src="https://image.tmdb.org/t/p/w300/nJVPgIq3icZMQ26p5uWIFdUZJfL.jpg"
    />
    <div
      className="onHover"
    >
      <button
        aria-label="Watch"
        className="btnWatch"
        onClick={[Function]}
        type="button"
      >
        <span>
           Watch Now 
        </span>
      </button>
      <button
        aria-label="View Info"
        className="btn ghost"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
    <div
      className="info"
    >
      <div>
        <h3
          className="title"
        >
          LelleBelle
        </h3>
        <p>
          <span
            className="genre"
          >
            Music
          </span>
          <span
            className="genre"
          >
            Romance
          </span>
        </p>
      </div>
      <span
        className="numRating"
      >
        4.4
      </span>
    </div>
  </article>,
]
`;
